crates/gpui/src/text_system.rs

```diff
@@ -78,18 +78,15 @@ impl TextSystem {
 
     /// Get a list of all available font names from the operating system.
     pub fn all_font_names(&self) -> Vec<String> {
-        let mut names: BTreeSet<_> = self
-            .platform_text_system
-            .all_font_names()
-            .into_iter()
-            .collect();
-        names.extend(self.platform_text_system.all_font_families());
+        let mut names = self.platform_text_system.all_font_names();
         names.extend(
             self.fallback_font_stack
                 .iter()
                 .map(|font| font.family.to_string()),
         );
-        names.into_iter().collect()
+        names.sort();
+        names.dedup();
+        names
```

Comment:
The reason for changing `BTreeSet` to `Vec` here is to avoid multiple memory allocations. However, if you prefer using `BTreeSet`, I can change it back. I'm not insisting on this change.